### PR TITLE
Fix pester tests syntax

### LINKED ISSUE #6265: Pester tests for Microsoft.PowerShell.Security are using null incorrectly
Several Pester tests for `Microsoft.PowerShell.Security` are incorrectly using `null` instead of `$null`

* Use of `Should not be null` should have used `$null`. Change to `Should -Not -BeNullOrEmpty` and remediate
  example: https://github.com/PowerShell/PowerShell/blob/76526c6f1df08c0a55fbae79a7148a4a58805626/test/powershell/Modules/Microsoft.PowerShell.Security/CertificateProvider.Tests.ps1#L51

* Use of `Get-Item` should be `Get-ChildItem` in cert provider test (#6258). Correcting `$null` check will expose this bug.
 https://github.com/PowerShell/PowerShell/blob/76526c6f1df08c0a55fbae79a7148a4a58805626/test/powershell/Modules/Microsoft.PowerShell.Security/CertificateProvider.Tests.ps1#L50

Discovered while working on #6256

### PR DESCRIPTION
**Pull request by xtqqczze**
# PR Summary

* Remove Unicode BOM from code files (#11546 was incomplete)
* Insert final newline into code files, as per `.vscode/settings.json`
* Clean up use of `BeOfType` parameter: remove unnecessary braces

## PR Context

* Follow-up to #11546

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [x] None
- **User-facing changes**
    - [x] Not Applicable
- **Testing - New and feature**
    - [x] N/A or can only be tested interactively
- **Tooling**
    - [x] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.

### CODE REVIEW

**Comment by xtqqczze** [comment]
@iSazonov PR is ready to merge, CodeFactor issues are not new.

**Comment by daxian-dbw** [comment]
> Pass 'BeOfType' parameter as [RuntimeType] instead of [string]

I don't changes related to this one are needed. `-BeOfType` takes a string as the type name, and this is how it's used in [Pester's sample code](https://github.com/pester/Pester/wiki/Should#beoftype).

**Comment by xtqqczze** [comment]
@daxian-dbw Judging from [BeOfType tests](https://github.com/pester/Pester/blob/master/Functions/Assertions/BeOfType.Tests.ps1), it is intended for `-BeOfType` to take a RuntimeType as well as a string. Apparently this is a documentation issue for pester.

Passing a RuntimeType preferable because it is constructed in the scope of the test rather than a pester function.

**Comment by iSazonov** [comment]
Usually we use `$p | Should -BeOfType [ActionPreference]`

**Comment by xtqqczze** [comment]
@iSazonov The parentheses are necessary for the parser to treat the token `[ActionPreference]` as an expression. Otherwise it is parsed as a string anyway:
 ```
$p | Should -BeOfType [ActionPreference]   # Fails
```

**Comment by daxian-dbw** [comment]
@xtqqczze I won't say either a RuntimeType or a string is preferable. Both are officially supported and both are useful. If you don't expect the type resolution to depend on anything from your script scope, then using a string is just fine. So I don't think it's necessary to change all `-BeOfType` instances to use a RuntimeType. We will not enforce this in our tests and gradually, it will again be a mix of both in our tests.

File: test/powershell/Modules/Microsoft.PowerShell.Security/CertificateProvider.Tests.ps1
Comment (iSazonov) [inline test/powershell/Modules/Microsoft.PowerShell.Security/CertificateProvider.Tests.ps1]: I think we should exclude this from the PR and make a fix as proposed in related issue.

**Review by iSazonov — APPROVED** [review]
LGTM with one comment.

File: test/powershell/Modules/Microsoft.PowerShell.Security/CertificateProvider.Tests.ps1
Comment (xtqqczze) [reply test/powershell/Modules/Microsoft.PowerShell.Security/CertificateProvider.Tests.ps1]: We cannot exclude this without fixing the issue. I would prefer to fix the underlying issue in a seperate PR.

**Comment by xtqqczze** [comment]
rebased to resolve merge confilcts

File: test/powershell/Modules/Microsoft.PowerShell.Security/CertificateProvider.Tests.ps1
Comment (iSazonov) [reply test/powershell/Modules/Microsoft.PowerShell.Security/CertificateProvider.Tests.ps1]: We should have working test after the _style_ change. If this is not so, we can not merge. Please revert the change and fix the behavior and the style in follow PR.

Comment (xtqqczze) [reply test/powershell/Modules/Microsoft.PowerShell.Security/CertificateProvider.Tests.ps1]: The test was broken before this PR so why should this block merge? I have WIP PR #11590 to fix the underlying issue but I am finding it hard to debug :(

Comment (iSazonov) [reply test/powershell/Modules/Microsoft.PowerShell.Security/CertificateProvider.Tests.ps1]: Our policy is (1) to split functional and style/formatting changes, (2) keep tests and code in right (working) state after any commit.
Before you change the test is passed so we should skip the "style" change and discuss in #11590.

Comment (xtqqczze) [reply test/powershell/Modules/Microsoft.PowerShell.Security/CertificateProvider.Tests.ps1]: @iSazonov The test is broken without this PR so it should be marked as pending in any case.

Maybe this helps to explain:
```powershell
$result = $null
$result | Should -Not -Be null  # passes
$result | Should -Not -Be $null # fails
```

File: test/powershell/Modules/Microsoft.PowerShell.Security/CertificateProvider.Tests.ps1
Comment (xtqqczze) [inline test/powershell/Modules/Microsoft.PowerShell.Security/CertificateProvider.Tests.ps1]: @iSazonov The test passes when `$null` is compared to `[string]'null'` when it should fail.

File: test/powershell/Modules/Microsoft.PowerShell.Security/CertificateProvider.Tests.ps1
Comment (iSazonov) [reply test/powershell/Modules/Microsoft.PowerShell.Security/CertificateProvider.Tests.ps1]: @xtqqczze After you revert the change we will merge and you can continue the work in #11590

**Review by iSazonov — APPROVED** [review]

**Pull request merged**
